src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -466,6 +467,33 @@ RequestContext<object> requestContext
             await requestContext.SendResult(result.First().ToString());
         }
 
+        private async Task HandleGetCommandsRequest(
+            object param,
+            RequestContext<object> requestContext)
+        {
+            var psCommand = new PSCommand();
+            psCommand.AddScript("Get-Command -CommandType Function, Cmdlet, ExternalScript | Sort-Object Name");
```

Comment:
If we think performance is a concern, we are better off sorting the result in C#. That should be possible with something like:

```csharp
var psCommand = new PSCommand()
    .AddScript("Get-Command -CommandType Function,Cmdlet,ExternalScript");

// Dunno if the <dynamic> bit works, but saves a coercion
var allCommands = await this.editorSession.PowerShellContext.ExecuteCommand<dynamic>(psCommand);

if (allCommands == null)
{
    await requestContext.SendResult(List<PSCommandMessage>.Empty);
    return;
}

var commandList = new List<PSCommandMessage>();
foreach (dynamic cmd in allCommands.OrderBy(c => c.Name, StringComparer.OrdinalIgnoreCase))
{
    commandList.Add(new PSCommandMessage()
        {
            Name = cmd.Name,
            ModuleName = cmd.ModuleName,
            Parameters = cmd.Parameters,
            ParameterSets = cmd.ParameterSets,
            DefaultParameterSet = cmd.DefaultParameterSet,
            CommandType = cmd.CommandType,
        });
}

await requestContext.SendResult(commandList);
```